### PR TITLE
[RabbitMQ] Add podManagementPolicy for cluster recovery.

### PR DESCRIPTION
**Pull request by ssup2**
Signed-off-by: Jungsub Shin <supsup5642@gmail.com>

#### What this PR does / why we need it:

rabbitmq should be initialized one by one when building cluster for the first time.
Therefore, the default value of podManagementPolicy is 'OrderedReady'

Once the rabbitmq participates in the cluster, it waits for a response from another
rabbitMQ in the same cluster at reboot, except the last rabbitmq of the same cluster.
If the cluster exits gracefully, you do not need to change the podManagementPolicy
because the first rabbitmq of the statefulset always will be last of the cluster.
However if the last rabbitmq of the cluster is not the first rabbitmq due to a failure,
you must change podManagementPolicy to 'Parallel'.
Otherwide the first rabbitmq waits Infinitely.

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/rabbitmq/values-production.yaml [hunk 1]
```diff
@@ -35,6 +35,18 @@ image:
...
+## rabbitmq should be initialized one by one when building cluster for the first time.
+## Therefore, the default value of podManagementPolicy is 'OrderedReady'
+## Once the rabbitmq participates in the cluster, it waits for a response from another 
+## rabbitMQ in the same cluster at reboot, except the last rabbitmq of the same cluster.
+## If the cluster exits gracefully, you do not need to change the podManagementPolicy 
+## because the first rabbitmq of the statefulset always will be last of the cluster.
+## However if the last rabbitmq of the cluster is not the first rabbitmq due to a failure, 
+## you must change podManagementPolicy to 'Parallel'.
+## ref : https://www.rabbitmq.com/clustering.html#restarting
+podManagementPolicy: OrderedReady
```
Comment (juan131) [inline stable/rabbitmq/values-production.yaml:49]: You used both `rabbitmq` and `rabbitMQ`. I'd use `RabbitMQ` for consistency. 

Please also follow the structure below for comments (include an empty comment before the parameter):

```
## Paramater description
## ref: https://xxxx
##
foo: bar

File: stable/rabbitmq/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: rabbitmq
-version: 5.1.0
+version: 5.1.1
```
Comment (juan131) [inline stable/rabbitmq/Chart.yaml]: Please increase minor version.

**Review by juan131 — CHANGES_REQUESTED** [review]
Hi @ssup2 

Thanks for your contribution! Please take a look to my comments when you have a chance.
Please also rebase your branch from master since it seems to have conflicts with in the Chart.yaml

File: stable/rabbitmq/values-production.yaml (hunk 1 above)
Comment (ssup2) [reply stable/rabbitmq/values-production.yaml:49]: Thanks for comments. I fixed it

File: stable/rabbitmq/Chart.yaml (hunk 2 above)
Comment (ssup2) [reply stable/rabbitmq/Chart.yaml]: I fixed it

**Review by tompizmor — APPROVED** [review]
/ok-to-test
/lgtm

**Comment by verwilst** [comment]
Changing podManagementPolicy to Parallel doesn't work:

```Error: StatefulSet.apps "rabbitmq-rabbitmq-ha" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden
Error: UPGRADE FAILED: StatefulSet.apps "rabbitmq-rabbitmq-ha" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden
```

**Comment by ssup2** [comment]
> Changing podManagementPolicy to Parallel doesn't work:
> 
> ```
> Error: UPGRADE FAILED: StatefulSet.apps "rabbitmq-rabbitmq-ha" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden
> ```

Oops. that is my mistake. As the above error message, statefulset doesn't support update the podManagementPolicy. The only way to update the podManagementPolicy is re-install chart via helm delete and helm install command with only change podManagementPolicy. As you know, helm delete command dosen't remove PVC. So if rabbitmq deployed with PVC, Queue and message is remained in PVC. 

I need to change comments in values.yaml. Thanks.

**Pull request merged**